### PR TITLE
Fix new_instances not working with Python 2.7

### LINKED ISSUE #50: Tests failing with Python 2.7 in 0.10.5
8 tests are failing in 0.10.5 version using Python 2.7. This seems to be caused by changes introduced in #40. I noticed the failing tests when I was fixing something else. All tests seem pass with Python 3. @christophe-riolo, do you have an idea with might be causing this?  

Running tests with ` ./tests/run_tests.sh`:
```
=========================================== test session starts ===========================================
platform linux2 -- Python 2.7.18, pytest-4.6.11, py-1.10.0, pluggy-0.13.1
rootdir: /home/ollipa/Projects/python/flexmock
collected 336 items                                                                                       

tests/flexmock_pytest_test.py ......................................................FF............. [ 20%]
............................................F.F.................................................... [ 50%]
............................................................................F...................F.. [ 79%]
.........................F.F.....................................xxxx                               [100%]

================================================ FAILURES =================================================
_______________ TestForPytest.test_flexmock_should_mock_new_instances_with_multiple_params ________________

self = <flexmock_pytest_test.TestForPytest object at 0x7fea6c854b10>

    def test_flexmock_should_mock_new_instances_with_multiple_params(self):
        class User(object):
            pass
    
        class Group(object):
            def __init__(self, arg, arg2):
                pass
    
        user = User()
        flexmock(Group).new_instances(user)
>       assert user is Group(1, 2)
E       TypeError: unbound method mock_method() must be called with Group instance as first argument (got type instance instead)

tests/flexmock_test.py:574: TypeError
___________________ TestForPytest.test_flexmock_should_revert_new_instances_on_teardown ___________________

self = <flexmock_pytest_test.TestForPytest object at 0x7fea6c6276d0>

    def test_flexmock_should_revert_new_instances_on_teardown(self):
        class User(object):
            pass
    
        class Group(object):
            pass
    
        user = User()
        group = Group()
        flexmock(Group).new_instances(user)
>       assert user is Group()
E       TypeError: unbound method mock_method() must be called with Group instance as first argument (got type instance instead)

tests/flexmock_test.py:586: TypeError
___________________________ TestForPytest.test_new_instances_should_be_a_method ___________________________

self = <flexmock_pytest_test.TestForPytest object at 0x7fea6c5a8f50>

    def test_new_instances_should_be_a_method(self):
        class Foo(object):
            pass
    
        flexmock(Foo).new_instances('bar')
>       assertEqual('bar', Foo())
E       TypeError: unbound method mock_method() must be called with Foo instance as first argument (got type instance instead)

tests/flexmock_test.py:1262: TypeError
___________________ TestForPytest.test_new_instances_works_with_multiple_return_values ____________________

self = <flexmock_pytest_test.TestForPytest object at 0x7fea6c4d9c90>

    def test_new_instances_works_with_multiple_return_values(self):
        class Foo(object):
            pass
    
        flexmock(Foo).new_instances('foo', 'bar')
>       assertEqual('foo', Foo())
E       TypeError: unbound method mock_method() must be called with Foo instance as first argument (got type instance instead)

tests/flexmock_test.py:1279: TypeError
_____________ TestUnittestClass.test_flexmock_should_mock_new_instances_with_multiple_params ______________

self = <flexmock_pytest_test.TestUnittestClass testMethod=test_flexmock_should_mock_new_instances_with_multiple_params>

    def test_flexmock_should_mock_new_instances_with_multiple_params(self):
        class User(object):
            pass
    
        class Group(object):
            def __init__(self, arg, arg2):
                pass
    
        user = User()
        flexmock(Group).new_instances(user)
>       assert user is Group(1, 2)
E       TypeError: unbound method mock_method() must be called with Group instance as first argument (got type instance instead)

tests/flexmock_test.py:574: TypeError
_________________ TestUnittestClass.test_flexmock_should_revert_new_instances_on_teardown _________________

self = <flexmock_pytest_test.TestUnittestClass testMethod=test_flexmock_should_revert_new_instances_on_teardown>

    def test_flexmock_should_revert_new_instances_on_teardown(self):
        class User(object):
            pass
    
        class Group(object):
            pass
    
        user = User()
        group = Group()
        flexmock(Group).new_instances(user)
>       assert user is Group()
E       TypeError: unbound method mock_method() must be called with Group instance as first argument (got type instance instead)

tests/flexmock_test.py:586: TypeError
_________________________ TestUnittestClass.test_new_instances_should_be_a_method _________________________

self = <flexmock_pytest_test.TestUnittestClass testMethod=test_new_instances_should_be_a_method>

    def test_new_instances_should_be_a_method(self):
        class Foo(object):
            pass
    
        flexmock(Foo).new_instances('bar')
>       assertEqual('bar', Foo())
E       TypeError: unbound method mock_method() must be called with Foo instance as first argument (got type instance instead)

tests/flexmock_test.py:1262: TypeError
_________________ TestUnittestClass.test_new_instances_works_with_multiple_return_values __________________

self = <flexmock_pytest_test.TestUnittestClass testMethod=test_new_instances_works_with_multiple_return_values>

    def test_new_instances_works_with_multiple_return_values(self):
        class Foo(object):
            pass
    
        flexmock(Foo).new_instances('foo', 'bar')
>       assertEqual('foo', Foo())
E       TypeError: unbound method mock_method() must be called with Foo instance as first argument (got type instance instead)

tests/flexmock_test.py:1279: TypeError
============================================ warnings summary =============================================
env/lib/python2.7/site-packages/_pytest/mark/structures.py:335
env/lib/python2.7/site-packages/_pytest/mark/structures.py:335: PytestUnknownMarkWarning: Unknown pytest.mark.pytest - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/latest/mark.html
    PytestUnknownMarkWarning,

-- Docs: https://docs.pytest.org/en/latest/warnings.html
======================= 8 failed, 324 passed, 4 xfailed, 1 warnings in 0.86 seconds =======================
```

### PR DESCRIPTION
**Pull request by ollipa**
Closes #50

### CODE REVIEW

File: flexmock.py [hunk 1]
```diff
@@ -847,7 +847,11 @@ def _update_method(self, expectation, name):
             if method_type in SPECIAL_METHODS:
                 expectation.original_function = getattr(obj, name)
             expectation.method_type = method_type
-        if not _isclass(obj) or expectation.method_type in SPECIAL_METHODS:
+        if (
+            not _isclass(obj)
+            or expectation.method_type in SPECIAL_METHODS
+            or name == '__new__'
```
Comment (christophe-riolo) [inline flexmock.py:853]: What would happen if we added __new__ to the special methods?

Comment (ollipa) [reply flexmock.py:853]: Special methods doesn't actually contain method names but types:
```python
SPECIAL_METHODS = (classmethod, staticmethod)
```

**Review by christophe-riolo — APPROVED** [review]
Manually tested it, all looks good! :+1: 
I just wonder why the new-style classes in python 2 did not behave as the python 3 classes :thinking:

**Comment by ollipa** [comment]
> Manually tested it, all looks good! 
> I just wonder why the new-style classes in python 2 did not behave as the python 3 classes

I'm not actually sure. I just observed from the tests that this happens with `__new__` method.

**Pull request merged**